packages/wonder-blocks-dropdown/components/action-menu.js

```diff
@@ -0,0 +1,237 @@
+// @flow
+// A menu that consists of action items
+
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import Button from "@khanacademy/wonder-blocks-button";
+
+import ActionItem from "./action-item.js";
+import DropdownCore from "./dropdown-core.js";
+import SelectItem from "./select-item.js";
+import SeparatorItem from "./separator-item.js";
+
+import type {
+    ActionItemProps,
+    SelectItemProps,
+    SeparatorProps,
+} from "../utils/types.js";
+
+type OpenerProps = {
+    /**
+     * Display text for the opener.
+     */
+    children: string,
+    /**
+     * Whether the opener is disabled. If disabled, disallows interaction.
+     */
+    disabled?: boolean,
+    /**
+     * Callback for when the opener is pressed.
+     */
+    onClick: () => void,
+};
+
+class ActionMenuOpener extends React.Component<OpenerProps> {
+    static defaultProps = {
+        disabled: false,
+    };
+
+    render() {
+        const {children, disabled, onClick} = this.props;
+
+        // TODO: incorporate caret once Icon is done
+        return (
+            <Button
+                role="menu"
+                color={"default"}
+                disabled={disabled}
+                kind={"tertiary"}
+                light={false}
+                onClick={onClick}
+                style={[styles.opener]}
+            >
+                {children}
+            </Button>
+        );
+    }
+}
+
+type ItemProps = ActionItemProps | SelectItemProps | SeparatorProps;
+
+type MenuProps = {
+    /**
+     * The items in this menu.
+     */
+    items: Array<ItemProps>,
+
+    /**
+     * Text for the opener of this menu.
+     */
+    menuText: string,
+
+    /**
+     * Whether this menu contains select items. If true, user should also define
+     * the onChange and selectedItems props.
+     */
+    containsSelectionItems?: boolean,
```

Comment:
This feels redundant.  Can't we assume `containsSelectionItems = Array.isArray(selectedValues)`?